stable_baselines/a2c/utils.py

```diff
@@ -1,297 +1,3 @@
-                    + _ln(tf.matmul(hidden, weight_h), gain_h, bias_h) + bias
-        else:
-            gates = tf.matmul(_input, weight_x) + tf.matmul(hidden, weight_h) + bias
-        in_gate, forget_gate, out_gate, cell_candidate = tf.split(axis=1, num_or_size_splits=4, value=gates)
-        in_gate = tf.nn.sigmoid(in_gate)
-        forget_gate = tf.nn.sigmoid(forget_gate)
-        out_gate = tf.nn.sigmoid(out_gate)
-        cell_candidate = tf.tanh(cell_candidate)
-        cell_state = forget_gate * cell_state + in_gate * cell_candidate
-        if layer_norm:
-            hidden = out_gate * tf.tanh(_ln(cell_state, gain_c, bias_c))
-        else:
-            hidden = out_gate * tf.tanh(cell_state)
-        input_tensor[idx] = hidden
-    cell_state_hidden = tf.concat(axis=1, values=[cell_state, hidden])
-    return input_tensor, cell_state_hidden
-
-
-def _ln(input_tensor, gain, bias, epsilon=1e-5, axes=None):
-    """
-    Apply layer normalisation.
-
-    :param input_tensor: (TensorFlow Tensor) The input tensor for the Layer normalization
-    :param gain: (TensorFlow Tensor) The scale tensor for the Layer normalization
-    :param bias: (TensorFlow Tensor) The bias tensor for the Layer normalization
-    :param epsilon: (float) The epsilon value for floating point calculations
-    :param axes: (tuple, list or int) The axes to apply the mean and variance calculation
-    :return: (TensorFlow Tensor) a normalizing layer
-    """
-    if axes is None:
-        axes = [1]
-    mean, variance = tf.nn.moments(input_tensor, axes=axes, keep_dims=True)
-    input_tensor = (input_tensor - mean) / tf.sqrt(variance + epsilon)
-    input_tensor = input_tensor * gain + bias
-    return input_tensor
-
-
-def lnlstm(input_tensor, mask_tensor, cell_state, scope, n_hidden, init_scale=1.0):
-    """
-    Creates a LSTM with Layer Normalization (lnlstm) cell for TensorFlow
-
-    :param input_tensor: (TensorFlow Tensor) The input tensor for the LSTM cell
-    :param mask_tensor: (TensorFlow Tensor) The mask tensor for the LSTM cell
-    :param cell_state: (TensorFlow Tensor) The state tensor for the LSTM cell
-    :param scope: (str) The TensorFlow variable scope
-    :param n_hidden: (int) The number of hidden neurons
-    :param init_scale: (int) The initialization scale
-    :return: (TensorFlow Tensor) lnlstm cell
-    """
-    return lstm(input_tensor, mask_tensor, cell_state, scope, n_hidden, init_scale, layer_norm=True)
-
-
-def conv_to_fc(input_tensor):
-    """
-    Reshapes a Tensor from a convolutional network to a Tensor for a fully connected network
-
-    :param input_tensor: (TensorFlow Tensor) The convolutional input tensor
-    :return: (TensorFlow Tensor) The fully connected output tensor
-    """
-    n_hidden = np.prod([v.value for v in input_tensor.get_shape()[1:]])
-    input_tensor = tf.reshape(input_tensor, [-1, n_hidden])
-    return input_tensor
-
-
 def discount_with_dones(rewards, dones, gamma):
```

Comment:
Is there a reason you did not move that one?

Reply:
This one was only imported by A2C so in that sense it was in its right place (and this is probably third-ish implementation of that whole function). It could be moved inside main a2c code though to get rid of this util file.